### ISSUE TITLE
Should `ShadowRootEventMap` have more events?

### ISSUE THREAD

**Issue by Fxxxlei** [microsoft/TypeScript-DOM-lib-generator#1527]
Events in a shadow DOM can bubble up to the shadow root and can be listened to on the `ShadowRoot` objects.

Thus I think `ShadowRootEventMap` should include more events, probably extending `GlobalEventHandlersEventMap`.

**Comment by saschanaz** [comment]
Sounds right, and I'd like to solve this in HTML spec instead (because ShadowRoot does not include GlobalEventHandlers right now). Can you open an issue there (and perhaps link back here)?

**Comment by AshleyScirra** [comment]
Just ran in to this. Events like "pointerdown" are allowed to be added to ShadowRoot and work correctly in practice, but are missing from ShadowRootEventMap, and so legal code can be incorrectly marked as an error. Example:

```ts
const shadowRoot = new ShadowRoot();
function handlePointerDown(e: PointerEvent) {}

// Error: Argument of type 'Event' is not assignable to parameter of type 'PointerEvent'
shadowRoot.addEventListener("pointerdown", e => handlePointerDown(e));
```

Workaround: write `handlePointerDown(e as PointerEvent)`.
